src/main/java/org/truffleruby/language/dispatch/UnresolvedDispatchNode.java

```diff
@@ -200,7 +201,8 @@ private CachedDispatchNode createMethodMissingNode(
 
                 if (!methodMissing.isDefined()) {
                     final String methodNameString = toString(methodName);
-                    throw new RaiseException(coreExceptions().noMethodErrorOnReceiver(methodNameString, receiverObject, argumentsObjects, this));
+                    throw new RaiseException(coreExceptions().noMethodError(ExceptionOperations.getFormatter(ExceptionOperations.NO_METHOD_ERROR, getContext()), receiverObject, methodNameString,
+                            argumentsObjects, this));
```

Comment:
Since the lines with `ExceptionOperations.getFormatter` are quite long, could you extract the `getFormatter` calls in local variables?